asv_bench/benchmarks/pandas/indexing.py

```diff
@@ -0,0 +1,296 @@
+        self.df.loc[self.tgt_scalar, :]
+
+    def time_loc_partial_key_bool_indexer(self, unique_levels):
+        self.df.loc[self.tgt_bool_indexer, :]
+
+    def time_loc_all_slices(self, unique_levels):
+        target = tuple([self.tgt_slice] * self.nlevels)
+        self.df.loc[target, :]
+
+    def time_loc_all_null_slices(self, unique_levels):
+        target = tuple([self.tgt_null_slice] * self.nlevels)
+        self.df.loc[target, :]
+
+    def time_loc_all_lists(self, unique_levels):
+        target = tuple([self.tgt_list] * self.nlevels)
+        self.df.loc[target, :]
+
+    def time_loc_all_scalars(self, unique_levels):
+        target = tuple([self.tgt_scalar] * self.nlevels)
+        self.df.loc[target, :]
+
+    def time_loc_all_bool_indexers(self, unique_levels):
+        target = tuple([self.tgt_bool_indexer] * self.nlevels)
+        self.df.loc[target, :]
+
+    def time_loc_slice_plus_null_slice(self, unique_levels):
+        target = (self.tgt_slice, self.tgt_null_slice)
+        self.df.loc[target, :]
+
+    def time_loc_null_slice_plus_slice(self, unique_levels):
+        target = (self.tgt_null_slice, self.tgt_slice)
+        self.df.loc[target, :]
+
+    def time_xs_level_0(self, unique_levels):
+        target = self.tgt_scalar
+        self.df.xs(target, level=0)
+
+    def time_xs_level_1(self, unique_levels):
+        target = self.tgt_scalar
+        self.df.xs(target, level=1)
+
+    def time_xs_full_key(self, unique_levels):
+        target = tuple([self.tgt_scalar] * self.nlevels)
+        self.df.xs(target)
+
+
+class GetItemSingleColumn:
+    def setup(self):
+        self.df_string_col = DataFrame(np.random.randn(3000, 1), columns=["A"])
+        self.df_int_col = DataFrame(np.random.randn(3000, 1))
+
+    def time_frame_getitem_single_column_label(self):
+        self.df_string_col["A"]
+
+    def time_frame_getitem_single_column_int(self):
+        self.df_int_col[0]
+
+
+class IndexSingleRow:
+    params = [True, False]
+    param_names = ["unique_cols"]
+
+    def setup(self, unique_cols):
+        arr = np.arange(10**7).reshape(-1, 10)
+        df = DataFrame(arr)
+        dtypes = ["u1", "u2", "u4", "u8", "i1", "i2", "i4", "i8", "f8", "f4"]
+        for i, d in enumerate(dtypes):
+            df[i] = df[i].astype(d)
+
+        if not unique_cols:
+            # GH#33032 single-row lookups with non-unique columns were
+            #  15x slower than with unique columns
+            df.columns = ["A", "A"] + list(df.columns[2:])
+
+        self.df = df
+
+    def time_iloc_row(self, unique_cols):
+        self.df.iloc[10000]
+
+    def time_loc_row(self, unique_cols):
+        self.df.loc[10000]
+
+
+class AssignTimeseriesIndex:
+    def setup(self):
+        N = 100000
+        idx = date_range("1/1/2000", periods=N, freq="H")
+        self.df = DataFrame(np.random.randn(N, 1), columns=["A"], index=idx)
+
+    def time_frame_assign_timeseries_index(self):
+        self.df["date"] = self.df.index
```

Comment:
Is this somehow particularly important to track?

Reply:
In essentially every case the answer is that "pandas implemented this for a reason"